### PR TITLE
Completed promise exercises

### PR DESCRIPTION
**Pull request by DariaBab**

### CODE REVIEW

File:  Promise-Exercises/script.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
+// Exercise 1: Basic Arithmetic Operations
+// Create a chain of promises to perform and log the result
+// of three arithmetic operations in sequence. Start with a number,
+// then add 5, multiply by 3, and finally subtract 2.
+
+const operationNumbers = {
```
Comment (Darkmift) [inline  Promise-Exercises/script.js:6]: nice approach!

File:  Promise-Exercises/script.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  multiplyNum: 3,
+  subtractNum: 2,
+};
+
+const mathPromise = new Promise((resolve) =>
+  resolve(operationNumbers.startNum)
+);
+
+mathPromise
+  .then((previousResult) => previousResult + operationNumbers.addNum)
```
Comment (Darkmift) [inline  Promise-Exercises/script.js:18]: good clear variable naming conventions

File:  Promise-Exercises/script.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+compareToNum(10, 5);
+
+// Exercise 4: Delayed Greetings
+// Simulate a delayed greeting with promises.
+// First, wait 2 seconds, then log "Hello", wait another second,
+// and log "World!". Each step should be done in a separate .then().
+
+const delayedPromise = () =>
```
Comment (Darkmift) [inline  Promise-Exercises/script.js:60]: good utility function

File:  Promise-Exercises/script.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  });
+
+// Exercise 5: Error Handling
+// Create a promise chain that attempts to parse JSON data.
+// Use a try/catch block within a .then() method to handle JSON parsing errors.
+// If successful, log the parsed object; if an error occurs, log "Invalid JSON".
+// Bonus
+// Make an async await version
+
+const JSONString =
```
Comment (Darkmift) [inline  Promise-Exercises/script.js:88]: alternatively you could declare an object and pass a stringified value

**Review by Avi-E-Koenig — APPROVED** [review]
Good clean code!

**Comment by DariaBab** [comment]
Avi, thank you so much for the feedback!🫶

вс, 7 апр. 2024 г. в 15:45, Avi E. Koenig ***@***.***>:

**Pull request merged**